doc/dev.md

```diff
@@ -48,3 +48,17 @@ Then run `node out/nbb_main.js <args>` to test the compiled nbb.
 ## Test
 
 To run tests, run `bb run-tests` for unit tests and `bb run-integration-tests` for running integration tests.
+
+## Features
+
+`nbb` can optionally bundle additional Clojure(Script) libraries as features. These can be specified with `$NBB_FEATURES` to compilation tasks e.g. `NBB_FEATURES=datascript,datascript-transit bb release`. The following features are provided:
+
+* [datascript](https://github.com/tonsky/datascript)
+* [datascript-transit](https://github.com/tonsky/datascript-transit)
+
+To add a new feature, add the following under `features/$LIBRARY/`:
+- `deps.edn` - dependencies for library
+- `shadow-cljs.edn` - compiler options necessary for advanced/release mode
+- `src/nbb/impl/$LIBRARY.cljs` - Sci mappings
+
+Then add entry/entries in `nbb.core` to require the library namespace(s).
```

Comment:
@logseq-cldwalker Perhaps we can register namespace data in some global map and then use that map to dispatch instead of case, for all the features. Then no code changes are necessary in core.